packages/cli-lib/lib/read.js

```diff
@@ -0,0 +1,67 @@
+const fs = require('fs');
+const path = require('path');
+
+const STAT_TYPES = {
+  FILE: 'file',
+  SYMBOLIC_LINK: 'symlink',
+  DIRECTORY: 'dir',
+};
+
+const generateFilePromise = (dir, file) => {
+  return new Promise((resolve, reject) => {
+    const filepath = path.join(dir, file);
+    fs.lstat(filepath, (error, stats) => {
+      if (error) {
+        reject(error);
+      }
+      let type;
+      if (stats.isSymbolicLink()) {
+        type = STAT_TYPES.SYMBOLIC_LINK;
+      } else if (stats.isDirectory()) {
+        type = STAT_TYPES.DIRECTORY;
+      } else if (stats.isFile()) {
+        type = STAT_TYPES.FILE;
+      }
+      resolve({ filepath, type });
+    });
+  });
+};
+
+const formatFilesData = filesData => {
+  return filesData.reduce((acc, fileData) => {
+    switch (fileData.type) {
+      case STAT_TYPES.FILE:
+        return acc.concat(fileData.filepath);
+      case STAT_TYPES.DIRECTORY:
+        return acc.concat(fileData.files || []);
+      case STAT_TYPES.SYMBOLIC_LINK:
+        // Skip symlinks
+        return acc;
+      default:
+        return acc;
+    }
+  }, []);
+};
+
+async function read(dir) {
+  const processFiles = files =>
+    Promise.all(files.map(file => generateFilePromise(dir, file)));
+
+  return new Promise((resolve, reject) => {
+    fs.readdir(dir, (error, files) => {
```

Comment:
Now that the minimum node version is 10, I think we could use https://nodejs.org/docs/latest-v10.x/api/fs.html#fs_fs_promises_api if we wanted. That said, if we are going to do more surgery, we should add test coverage.

Reply:
updated in b12426c, but I still need to add tests for `read` and `write`. I'm not quite sure what the best way to mock `fs` is. I'll look into it though.

Reply:
Maybe some inspiration in https://github.com/HubSpot/hubspot-cli/blob/master/packages/cli-lib/__mocks__/fs-extra.js?